cli/src/main/kotlin/commands/AnalyzerCommand.kt

```diff
@@ -173,12 +174,20 @@ class AnalyzerCommand : CliktCommand(name = "analyze", help = "Determine depende
 
         val repositoryConfiguration = actualRepositoryConfigurationFile?.readValueOrNull() ?: RepositoryConfiguration()
 
+        val packageCurations =
+            FilePackageCurationProvider.from(packageCurationsFile, packageCurationsDir).packageCurations.toMutableList()
```

Comment:
How about using `.toMutableSet()` here and omitting `.toSet()` in line 190 below?

Reply:
AFAIK, a Set does not guarantee to retain the order of the elements, but this is required to ensure that the local curations will take precedence over the global ones.

See Line 183:
```kt
if (config.applyRepositoryPackageCurations) {
    packageCurations += repositoryPackageCurations
} else if (repositoryPackageCurations.isNotEmpty()) {
    log.warn { "Local package-curation were not applied because the feature is not enabled." }
}
```

Reply:
For a general set you're right, but `toMutableSet()` guarantees that "The returned set preserves the element iteration order of the original collection."

Reply:
Yes in the initial transformation to a mutable map.
When adding other elements, which is being done in Line 183, the order is no longer guaranteed, right?

Reply:
> When adding other elements, which is being done in Line 183, the order is no longer guaranteed, right?

Hmm, you're probably right, good point.